api/PclusterApiHandler.py

```diff
@@ -582,27 +582,29 @@ def get_identity():
         return {"user_roles": ["user", "admin"], "username": "username", "attributes": {"email": "user@domain.com"}}
 
     access_token = request.cookies.get("accessToken")
-    id_token = request.cookies.get("idToken")
-    if not (access_token and id_token):
-        return {"message": "No access or id token."}, 401
-    try:
-        claims = ["email"]
+    id_token = request.cookies.get("idToken", None)
 
+    claims = ["email"]
+    try:
         decoded_access = jwt_decode(access_token)
-        identity_from_access_token = _get_identity_from_token(decoded=decoded_access, claims=claims)
+    except jwt.ExpiredSignatureError:
+        if not auth_cookies:
+            return {}, 401
+
+        decoded_access = jwt_decode(auth_cookies['accessToken'])
```

Comment:
question: why is this fallback mechanism needed?

Reply:
strictly speaking this is not actually needed, since it's taken care of by the `@authenticated` decorator
I put it as a defensive approach and also an explanatory action to the reader
I'm open to remove it !

Reply:
On a second thought, I decided to remove it 
Thanks!